### ISSUE TITLE
Session keyword argument guidance is misleading when using a required keyword argument

### ISSUE THREAD

**Issue by aaronjensen** [eventide-project/messaging#11]
https://github.com/eventide-project/messaging/blob/dda5a9880028ad002d8fdfb89d7f2ed8558b5838/lib/messaging/handle.rb#L55-L57

Required keyword arguments have the type `:keyreq`, so if you use one, you get a message telling you to use a keyword argument. Also, if you use a positional argument, you get a message telling you to use a keyword argument (but it does not specify optional or required). 

Ideally the message would guide you in the right direction or would accept either optional or required.

**Comment by ntl** [comment]
Agreed about the ambiguity, but does using a required keyword argument on an applicative handler's configure method ever make optimal sense? The configure method is generally called by `.build` on the class, which may or may not pass in a session. Therefore, making the session argument to `#configure` a required one doesn't have any effect on the actual construction interface offered by the handler. In the typical operational context, the user is assured that the session argument will always be supplied by the dispatching consumer.

Maybe the error message should simply add a bit of clarification: `Optional session parameter of configure is not an optional keyword argument`

**Comment by aaronjensen** [comment]
I don't know if it makes sense or not, which is why I took a guess. As far as I'm concerned, I require it (though, of course, it could be `nil`). Build on the class always passes a session in (again, it could be `nil`, but `nil` is still passed afaict) in if I'm using the one I get by including `Messaging::Handle`. I'm not sure what wouldn't pass it in, so I'm not sure how to think about it completely.

I'm good w/ either clarification or permissiveness.

**Comment by ntl** [comment]
Talked it over with @sbellware , we're going to change the handler to permit the `session` to be a required keyword argument on `configure`.

It will have unsurprising consequences for anyone thinking that making it a required argument somehow eliminates the possibility of the session value ever being nil.

**Comment by aaronjensen** [comment]
Sounds good, thanks!

**Comment by sbellware** [comment]
It will only be a footnote in the documentation, rather than conveyed as a primary use case.

**Comment by aaronjensen** [comment]
Works for me, I wouldn't even expect it to be documented.

**Comment by sbellware** [comment]
👍
